plasoscaffolder/frontend/frontend.py

```diff
@@ -0,0 +1,373 @@
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      list: the answer as supplied by the user.
+    """
+    return_list = []
+
+    more_entries = True
+    entry_index = 1
+    while more_entries:
+      # pylint: disable=assignment-from-no-return
+      value = cls.OUTPUT_HANDLER.PromptInfo(
+          'Value to add [#{0:d}]'.format(entry_index))
+      if not value:
+        cls.OUTPUT_HANDLER.PrintError('Empty value, not adding.')
+      else:
+        return_list.append(value)
+
+      more_entries = cls.OUTPUT_HANDLER.Confirm('Add more values?', abort=False)
+      entry_index += 1
+
+    question.ValidateAnswer(return_list)
+    return return_list
+
+  @classmethod
+  def _AskQuestion(cls, question: scaffolder_interface.BaseQuestion) -> Type:
+    """Ask the user a question and return an answer back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      object: an object, whos type is defined in question.type.
+
+    Raises:
+      errors.WrongCliInput: when an unsupported question type is
+          encountered.
+    """
+    cls.OUTPUT_HANDLER.PrintNewLine()
+    cls.OUTPUT_HANDLER.PrintInfo(question.prompt)
+
+    if question.TYPE == str:
+      return cls._AskStringQuestion(question)
+
+    if question.TYPE == dict:
+      return cls._AskDictQuestion(question)
+
+    if question.TYPE == list:
+      return cls._AskListQuestion(question)
+
+    # TODO: Add support for other types of questions.
+    raise errors.WrongCliInput(
+        'Question type {0:s} not supported.'.format(str(question.TYPE)))
+
+  @classmethod
+  def _AskStringQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> str:
+    """Ask the user a question and return a string answer back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      str: the answer as supplied by the user.
+    """
+    # pylint: disable=assignment-from-no-return
+    value = cls.OUTPUT_HANDLER.PromptInfo('Value')
+    question.ValidateAnswer(value)
+    return value
+
+  @classmethod
+  def _GetSelection(cls, items: list, item_text: str) -> str:
+    """Present a list of items to user and return back the chosen one.
+
+    Args:
+      items (list): the list of items to present to the user.
+      item_text (str): heading used in presentation to the user.
+
+    Returns:
+      str: the chosen item from the list.
+    """
+    for item_count, item in enumerate(items):
+      cls.OUTPUT_HANDLER.PrintInfo(
+          '  [{0:d}] {1:s}'.format(item_count + 1, item))
+
+    # pylint: disable=assignment-from-no-return
+    result = cls.OUTPUT_HANDLER.PromptInfo('{0:s} choice'.format(item_text))
+    try:
+      result_int = int(result, 10)
+
+      if result_int <= len(items):
+        return items[result_int - 1]
+    except ValueError:
+      pass
+
+    return result
```

Comment:
So if the user enters "100" when there's only 10 options, we just return it? Why?